### PR TITLE
Added support for new fmi export flags

### PR DESCRIPTION
**Pull request by adeas31**
Fixes ticket:3105 and ticket:5969

### CODE REVIEW

File: OMCompiler/Compiler/Util/Flags.mo [hunk 1]
```diff
@@ -1355,7 +1355,12 @@ constant ConfigFlag FLAT_MODELICA = CONFIG_FLAG(141, "flatModelica",
   Gettext.gettext("Outputs experimental flat Modelica."));
 constant ConfigFlag FMI_FILTER = CONFIG_FLAG(142, "fmiFilter", NONE(), EXTERNAL(),
   ENUM_FLAG(FMI_INTERNAL, {("none", FMI_NONE), ("internal", FMI_INTERNAL), ("protected", FMI_PROTECTED), ("blackBox", FMI_BLACKBOX)}),
-  SOME(STRING_OPTION({"none", "internal", "protected", "blackBox"})),
+  SOME(STRING_DESC_OPTION({
+    ("none", Gettext.gettext("All variables will be exposed, even variables that are introduced by the symbolic transformations. Hence, this is intended to be used for debugging.")),
+    ("internal", Gettext.gettext("All internal variables introduced by the symbolic transformations are filtered out. Only the variables from the actual Modelica model are exposed (with minor exceptions, e.g. for state sets).")),
```
Comment (arun3688) [inline OMCompiler/Compiler/Util/Flags.mo:1360]: "internal"  to  Gettext.gettext("Default Option, All internal variables introduced by the symbolic transformations are filtered out. Only the variables from the actual Modelica model are exposed (with minor exceptions, e.g. for state sets)."))

Comment (adeas31) [reply OMCompiler/Compiler/Util/Flags.mo:1360]: Not sure if its a good idea to write this in the text. Users guide automatically read it from the default option from the default argument of flag. https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omchelptext.html#omcflag-fmifilter

Comment (arun3688) [reply OMCompiler/Compiler/Util/Flags.mo:1360]: Then it is fine

**Comment by lochel** [comment]
Now I can see the tooltips. For some reason I had to recompile to make them show up.

**Comment by lochel** [comment]
Do we need to update the documentation [omedit.html#fmi](https://www.openmodelica.org/doc/OpenModelicaUsersGuide/latest/omedit.html#fmi) manually or does it get automatically generated?

**Comment by lochel** [comment]
The option doesn't get disabled in my build. But I am afraid that there is a problem on my side...

**Comment by adeas31** [comment]
Make sure `main.cpp` is compiled again.

**Comment by lochel** [comment]
It worked now. However, the build system should compile `main.cpp` if needed.

**Comment by adeas31** [comment]
> It worked now. However, the build system should compile `main.cpp` if needed.

See #903

**Pull request merged**